src/openrct2-ui/windows/Staff.cpp

```diff
@@ -768,31 +793,28 @@ namespace OpenRCT2::Ui::Windows
             }
 
             int32_t checkedIndex = -1;
-            // This will be moved below where Items Checked is when all
-            // of dropdown related functions are finished. This prevents
-            // the dropdown from not working on first click.
-            int32_t numCostumes = StaffGetAvailableEntertainerCostumeList(_availableCostumes);
-            for (int32_t i = 0; i < numCostumes; i++)
-            {
-                EntertainerCostume costume = _availableCostumes[i];
-                if (staff->AnimationGroup == EntertainerCostumeToSprite(costume))
-                {
+            auto numCostumes = _availableCostumes.size();
+            for (auto i = 0u; i < numCostumes; i++)
+            {
+                // TODO: refactor this hack
```

Comment:
How and when?

Reply:
Currently, dropdown args are defined as an `int64_t`. This means the `char*` is assigned using `memcpy`, so it works with both 32-bit and 64-bit builds. It feels incredibly hacky to me, but it is how it works in several places in the codebase.

Ideally, there'd be a more universal (variant) interface for dropdown arguments. I'll amend the comment to clarify.